cpp/src/sampling/neighbor_sampling_impl.hpp

```diff
@@ -150,7 +173,7 @@ neighbor_sample_impl(
 
   std::vector<size_t> level_sizes{};
   int32_t hop{0};
-  for (auto&& k_level : fan_out) {
+  for (auto&& k_level : (*fan_out)) {
```

Comment:
This isn't actually sufficient yet... but I'm more worried about the API right now.

This loop will need, in the case of heterogeneous sampling, to have 2 levels of for loop.  An outer loop iterating by hop and an inner loop iterating by type.

I'd be inclined to add a setup loop that iterates over the types and generates the masks - and perhaps identifies the maximum number of hops to drive the outer loop.  You'll need to get k_level from the right type/hop combination... so this for construct won't work at all, it will need to look different.

Reply:
Right I only added it for it to compile. I will revisit this approach once we lock the API's interface